homeassistant/components/geo_location/nsw_rural_fire_service_feed.py

```diff
@@ -0,0 +1,245 @@
+HA instance's location.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/geo_location/nsw_rural_fire_service_feed/
+"""
+import logging
+from datetime import timedelta
+from typing import Optional
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.geo_location import GeoLocationEvent, \
+    PLATFORM_SCHEMA
+from homeassistant.const import CONF_RADIUS, CONF_SCAN_INTERVAL, \
+    EVENT_HOMEASSISTANT_START, ATTR_ATTRIBUTION
+from homeassistant.helpers.event import track_time_interval
+
+REQUIREMENTS = ['geojson_client==0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_CATEGORY = 'category'
+ATTR_COUNCIL_AREA = 'council_area'
+ATTR_EXTERNAL_ID = 'external_id'
+ATTR_FIRE = 'fire'
+ATTR_LOCATION = 'location'
+ATTR_PUBLICATION_DATE = 'publication_date'
+ATTR_RESPONSIBLE_AGENCY = 'responsible_agency'
+ATTR_SIZE = 'size'
+ATTR_STATUS = 'status'
+ATTR_TYPE = 'type'
+
+CONF_CATEGORIES = 'categories'
+
+DEFAULT_RADIUS_IN_KM = 20.0
+DEFAULT_UNIT_OF_MEASUREMENT = "km"
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+VALID_CATEGORIES = ['Emergency Warning', 'Watch and Act', 'Advice',
+                    'Not Applicable']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_RADIUS, default=DEFAULT_RADIUS_IN_KM):
+        vol.Coerce(float),
+    vol.Optional(CONF_CATEGORIES, default=[]):
+        vol.All(cv.ensure_list, [vol.In(VALID_CATEGORIES)])
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the GeoJSON Events platform."""
+    scan_interval = config.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL)
+    radius_in_km = config[CONF_RADIUS]
+    categories = config.get(CONF_CATEGORIES)
+    # Initialize the entity manager.
+    NswRuralFireServiceFeedManager(hass, add_entities, scan_interval,
+                                   radius_in_km, categories)
+
+
+class NswRuralFireServiceFeedManager:
+    """Feed Manager for NSW Rural Fire Service GeoJSON feed."""
+
+    def __init__(self, hass, add_entities, scan_interval, radius_in_km,
+                 categories):
+        """Initialize the GeoJSON Feed Manager."""
+        from geojson_client.nsw_rural_fire_service_feed \
+            import NswRuralFireServiceFeed
+        self._hass = hass
+        self._feed = NswRuralFireServiceFeed((hass.config.latitude,
+                                              hass.config.longitude),
+                                             filter_radius=radius_in_km,
+                                             filter_categories=categories)
+        self._add_entities = add_entities
+        self._scan_interval = scan_interval
+        self._feed_entries = []
+        self._managed_entities = []
+        hass.bus.listen_once(
+            EVENT_HOMEASSISTANT_START, lambda _: self._update())
+        self._init_regular_updates()
```

Comment:
We should do this at HA start event callback

Reply:
OK, moving into `async_added_to_hass`, too

Reply:
Same as above - going to move into call from `setup_platform` instead.